File: 백준/Bronze/14758. Paint Me/Paint Me.py
import sys
input=lambda:sys.stdin.readline().rstrip()
while True:
    n,w,l,h,a,m=map(int,input().split())
    if n==0: break
    total=w*h*2+h*l*2+w*l
    for _ in range(m):
        x,y=map(int,input().split())
        total-=x*y
    print(total*n//a+(1 if total*n%a!=0 else 0))

File: 백준/Bronze/14758. Paint Me/Paint Me.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, w, l, h, a, m, x, y, total;
    while (true) {
        cin >> n >> w >> l >> h >> a >> m;
        if (n == 0) break;
        total = w * h * 2 + h * l * 2 + w * l;
        for (int i = 0; i < m; i++) {
            cin >> x >> y;
            total -= x * y;
        }
        cout << total * n / a + (total * n % a == 0 ? 0 : 1) << "\n";
    }
    return 0;
}